### ISSUE TITLE
For Structured Chat Agent, GPT-4 often responds directly.

### ISSUE THREAD

**Issue by Sherv33n** [langchain-ai/langchain#4059]
When utilizing [a Structured Chat Agent](https://github.com/hwchase17/langchain/pull/3912), GPT-4 will often send a direct response when it should be crafting a JSON blob.

As an example, I prompted a Playwright-driven agent w/ `Can you summarize this site: ramp.com?`

The first gpt-4 agent response is:
``````
ASSISTANT
Action:
```
{
  "action": "navigate_browser",
  "action_input": "https://ramp.com"
}
```
``````

And, after navigating to the page, gpt-4 responds:
```
ASSISTANT
I need to extract the text from the website to summarize it.
```

Whereas gpt-3.5-turbo responds with:
``````
ASSISTANT
To summarize the site ramp.com, I can extract the text on the webpage using the following tool:
Action:
```
{
  "action": "extract_text",
  "action_input": {}
}
```
``````

In other words, GPT-4 seems to take `Respond directly if appropriate.` from the system prompt too loosely.

**Comment by vowelparrot** [comment]
Thanks for reporting this issue! We will look into issues of hallucination for this agent.  Are you able to get a prompt that works better for your use case?

**Comment by pierreembacher** [comment]
Here something that worked for me: ""Look for (..) and summarize but finish chain with that summery and NOT earlier."

**Comment by pierreembacher** [comment]
"At the Flüchtlingsgipfel, German federal and state governments are in a long-standing dispute over the distribution of costs for accommodating refugees. The states demand more financial support from the federal government, while the latter believes it is already contributing disproportionately to the costs. The states criticize the federal government for not providing enough funds for language courses, additional daycare groups, and more integration services for families and children. The federal government has already provided €15.6 billion to the states this year. The meeting is not expected to result in a long-term solution, but a temporary agreement for the current year would be considered a success." The output was way better with gpt-4.

**Comment by rheaton64** [comment]
I'm having this same issue. It occurs with gpt4, but not gpt3.5. I've also found that the higher the temperature, the more likely the gpt4 agent is to complete the task instead of finishing early. A temp of 1 seems to sometimes work, while a temp of 0 will never work correctly. I'll experiment with some changes to the prompt and see what I can find

**Comment by indiejoseph** [comment]
What is the different between ChatAgent and StructedChatAgent? their output format also json blob, and having very similar instruction, I can see the Final Answer is in different format and StructedChatAgent support tool args, but when to use StructedChatAgent over ChatAgent? if the main purpose is to support tool args, why not directly modify the ChatAgent? 

btw does anyone tested is the hallucination occurring with ChatAgent?

**Comment by OlajideOgun** [comment]
+1

**Comment by OlajideOgun** [comment]
When adding in memory to structured_chat_agent it completely ignores tools

**Comment by okradze** [comment]
Same issue
